### PR TITLE
Feature/functions milestone 3

### PR DESCRIPTION
**Pull request by AymanJabr**
In this branch, I created the calculate and operate functions, which together should be able to take in any valid operation/number, and give back the result of the operation.

The calculate function takes in 2 parameters: calculatorData, and operationName
It returns an object with the result of the passed in calculatorData values.

The operate function takes in 3 parameters: numberOne, numberTwo, and operation. 
It returns the result of using the operation on numberOne and numberTwo

### CODE REVIEW

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+export default function calculate(calculatorData, buttonName) {
+  let myNext = calculatorData.next;
+  let myOperation = calculatorData.operation;
+  let myTotal = calculatorData.total;
```
Comment (Berabjesus) [inline src/logic/calculate.js]: - [x] use ES6 object destructuring 
```
  let { total, next, operation } = calculatorData;

```

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  const bigOne = new Big(numberOne);
+  const bigTwo = new Big(numberTwo);
+
+  switch (operation) {
+    case '+':
+      return bigOne + bigTwo;
+    case '-':
+      return bigOne - bigTwo;
+    case '/':
+      return bigOne / bigTwo;
```
Comment (Berabjesus) [inline src/logic/operate.js]: - [x] should check if `bigTwo` is zero and handle the error.

**Review by Berabjesus — CHANGES_REQUESTED** [review]
## Status: REQUIRED CHANGES :warning: :construction: 
Hi @AymanJabr great start to this milestone 👏 
### you have a few changes to carry out before we proceed so please take a look at the things you have to work on and their respective review comments below.

- [x] `calculate.js` must import `operate.js`.
- [x] Use ES6 object destructuring `calculate.js`. Please check the inline comments below for more comments
- [x] operate.js should check for zero divisions. Please check the inline comments below for more comments

#### **If you have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help** 
**HAPPY CODING :computer:[Bereket Beshane](https://bereketababu.me)** | [LinkedIn](https://linkedin.com/in/bereketbeshane) | [GitHub](https://github.com/Berabjesus)

**Review by Temmarie — APPROVED** [review]
## Status: Approved :heavy_check_mark:

### Hi @AymanJabr  ! :wave:

#### Thank you for making all of the required changes, your project is approved :clap: :clap: :clap: :tada:
#### You may merge, good luck with your next projects! :rocket:
#### Happy Coding! :+1:
__Grace Tamara Ekunola __

**Pull request merged**